app/controllers/incomeSources/add/IncomeSourceReportingMethodController.scala

```diff
@@ -83,41 +83,43 @@ class IncomeSourceReportingMethodController @Inject()(val authorisedFunctions: F
       handleRequest(isAgent = isAgent, incomeSourceType)
   }
 
-  def handleRequest(isAgent: Boolean, incomeSourceType: IncomeSourceType)
-                   (implicit user: MtdItUser[_]): Future[Result] = withSessionData(JourneyType(Add, incomeSourceType), journeyState = AfterSubmissionPage) { _ =>
-
-    sessionService.getMongoKeyTyped[String](AddIncomeSourceData.incomeSourceIdField, JourneyType(Add, incomeSourceType)).flatMap {
-      case Right(Some(id)) =>
-        handleIncomeSourceIdRetrievalSuccess(incomeSourceType = incomeSourceType, id = id, isAgent = isAgent)
-      case Right(_) => Future.failed(new Error("[IncomeSourceReportingMethodController][handleSubmit] Could not find an incomeSourceId in session data"))
-      case Left(ex) => Future.failed(ex)
+  def handleRequest(isAgent: Boolean, incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_]): Future[Result] = {
+    withSessionData(JourneyType(Add, incomeSourceType), journeyState = AfterSubmissionPage) { sessionData =>
+
+      val incomeSourceId = sessionData.addIncomeSourceData.flatMap(_.incomeSourceId)
```

Comment:
redundant variable declaration, we can pattern match over flatMap result right away